reactive/cwr.py

```diff
@@ -107,6 +107,12 @@ def install_jenkins_jobs(connected_jenkins):
         if not installed:
             hookenv.log("installation of {} did not complete on time."
                         .format(plugin))
+            # Retrying does not play well here.
+            # I have seen a case where Jenkins mistakenly reports that
+            # the plugin is installed causing an infinite retry loop :(
+            # Here we report that jenkins failed and we block.
+            set_state("jenkins.jobs.failed")
```

Comment:
This state is not to be removed. The user should re-add the cwr-jenkins relation. However, note that even re-adding the relation might not resolve the issue of failing to deploy plugins. I have seen a case where Jenkins was saying the plugin is deployed but in reality it was not. Such cases may require human to figure out what went wrong and perhaps manually add/remove plugins.

This PR does not try to do anything smart in respect to plugins. Here we report the failure without erroring and completely blocking the deployment. Being smart in case of failures and trying to remove/add plugins needs more investigation and should be handled in another PR.